File: StoryAdExample/ObjcImplementation/ObjcImplementation/ViewController.m
//
//  ViewController.m
//  ObjcImplementation
//
//  Created by Rishabh Tripathi on 13/06/25.
//

#import "ViewController.h"

@interface ViewController () <ApStoryDelegate>

@property (strong, nonatomic) ApStory *apStory;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.apStory = [[ApStory alloc] init];
}

-(IBAction)loadAd:(UIButton *)sender {
    if (self.apStory) {
        [self.apStory loadAdWithPlacementId:@"testNativeAdUnit" swipeUpGestureEnabled:false clickDismissEnabled:false delegate:self];
    }
}

-(IBAction)showAd:(UIButton *)sender {
    if (self.apStory) {
        UIView *adView = [self.apStory getAd];
        [self.view addSubview:adView];
        adView.translatesAutoresizingMaskIntoConstraints = NO;

        [NSLayoutConstraint activateConstraints:@[
            [adView.leadingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.leadingAnchor],
            [adView.trailingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.trailingAnchor],
            [adView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
            [adView.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor]
        ]];
    }
    
}

- (void)onComplete:(BOOL)success {
    NSLog(@"Ad Complete");
}

@end

File: StoryAdExample/SwiftImplementation/SwiftImplementation/ViewController.swift
//
//  ViewController.swift
//  SwiftImplementation
//
//  Created by Rishabh Tripathi on 10/06/25.
//

import UIKit
import ApMobileSDK

class ViewController: UIViewController, ApStoryDelegate {
    let apStory = ApStory()

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }
    
    @IBAction func loadAd(_ sender: UIButton) {
        apStory.loadAd(placementId: "testNativeAdUnit", swipeUpGestureEnabled: false, clickDismissEnabled: false, delegate: self)
    }

    @IBAction func showAd(_ sender: UIButton) {
        let adView = apStory.getAd()
        self.view.addSubview(adView)
        adView.translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([
            adView.leadingAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.leadingAnchor),
            adView.trailingAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.trailingAnchor),
            adView.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
            adView.bottomAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.bottomAnchor)])
    }

    func onComplete(_ success: Bool) {
        print("Debug Ad Complete")
    }
    
    func onAdLoaded() {
        print("Ad Loaded")
    }
    
    func onError(_ code: Int, _ message: String) {
        print("Error: \(code) \(message)")
    }
}

